jekyll/_cci2/vs-code-extension-overview.adoc

```diff
@@ -11,34 +11,161 @@ contentTags:
+
+[#config-helper]
+== Config helper
+
+The VS Code extension also provides in-file contextual help for your CircleCI configs, including:
+
+- **Rich code navigation through “go-to-definition” and “go-to-reference” commands.** Hover over a job name or executor parameter to verify its definition or view where they are referenced elsewhere in the file. You can also explore the definition of orb-defined commands or parameters.
++
+image::{{site.baseurl}}/assets/img/docs/vs_code_extension_config_helper_go-to-definition-optimised.gif[Screenshot showing the definition available on hover]
+
+- **Contextual documentation and usage hints when hovering on specific keys.** This helps you avoid having to frequently switch to your browser to check the documentation when editing your configuration. Links to the official CircleCI docs are also provided on hover, for easier navigation.
++
+image::{{site.baseurl}}/assets/img/docs/vs_code_extension_config_helper_on-hover-documentation.png[Screenshot showing the contextual information on hover]
+
+- **Syntax validation**. This helps you identify typos, incorrect use of parameters, incomplete definitions, wrong types, invalid or deprecated machine versions, etc.
++
+image::{{site.baseurl}}/assets/img/docs/vs_code_extension_config_helper_syntax-validation.gif[Screenshot showing the synthax highlightning when an error is identified]
+
+- **Usage warnings**. This helps you identify deprecated parameters, unused jobs or executors, or missing properties that prevent you from taking advantage of CircleCI’s full capabilities.
++
+image::{{site.baseurl}}/assets/img/docs/vs_code_extension_config_helper_usage-warning.png[Screenshot showing code highlightning to warn on an unused job]
+
+- **Auto completion**. This is available with both built-in keys and parameters as well as user-defined variables.
++
+image::{{site.baseurl}}/assets/img/docs/vs_code_extension_config_helper_autocomplete.png[Screenshot showing two suggestions to autocomplete the line of code]
+
+[#config-validation-commands]
+=== Config validation commands
+
+The config helper also provides two commands that help you statically validate your YAML config files without having to run a pipeline.
+
+* Validate current configuration file
++
+Corresponds to the CLI command `circleci config validate`, and verifies statically that the config file is well formed. Please note that this command only validates this file for structure and syntax errors, but not for semantic error (e.g. "This job does not exist").
+
+* Validate current configuration file against org policy
++
+Corresponds to the CLI command `circleci policy decide`, and verifies that the configuration file complies with your organisation policies (if any are set).
+
+Both of these commands can be invoked by:
+
+- Right-clicking on a CircleCI YAML file.
```

Comment:
Not sure how to do this option

Reply:
Ah, it's actually **within** the open file in the editor

https://user-images.githubusercontent.com/36839689/232631817-2be65134-eb7e-4064-b333-c768c1fa5edd.mov

@BeFunes In this screen capture I have two "CircleCI > Validate..." options when I right click presumably because there are two config.yml files in this workspace, is that right? (One for the docs-platform repo and one in __content which is a git submodule here)

Reply:
Relatedly, I do still see the CircleCI button on the top right even if I am editing a different file

Reply:
@BeFunes I also just saw that the validate commands are available from the Command Palette, I think it would be a good idea to add that?

Reply:
Oh thanks Kat, don't know how I missed that! 🙏